internal/sqlquery/row_scanner.go

```diff
@@ -32,7 +32,7 @@ func newRowScanner(colTypes []colType) *rowScanner {
 			}
 			format := "%v"
 			if t, isTime := v.(time.Time); isTime {
-				return t.Format(time.RFC3339), nil
+				return t.Format(time.RFC3339Nano), nil
```

Comment:
Is this a breaking change? From what you're describing, it looks like this bug only impacts users when timestamps are in `RFC3339Nano`, but it works as expected when timestamps are in `RFC3339`. Is that correct?

Is there some way to test which format the timestamp is in, and format accordingly? Or is there a reason we should always use `RFC3339Nano` instead of `RFC3339`?

Reply:
> Is this a breaking change? 

It's not a breaking change - I marked this PR as `bug_fix` in the changelog. My other PR #35189 - is a breaking change.

> From what you're describing, it looks like this bug only impacts users when timestamps are in RFC3339Nano, but it works as expected when timestamps are in RFC3339. Is that correct?

If by timestamp in `RFC3339` format you mean that timestamp records in database don't have millisecond precision (ex. `2022-06-03 21:59:26.000000+00`, `2022-06-03 21:59:31.000000`) - then yes, error would not appear. 

But just by type inspection we can't know if client used `.000000` intentionally to avoid MS precision, or it's intended point in time.

> Is there some way to test which format the timestamp is in, and format accordingly? Or is there a reason we should always use RFC3339Nano instead of RFC3339?

`RFC3339Nano` is safer, `RFC3339` causes data loss for timestamp.

Reply:
I'm pretty ignorant on timestamp formats, so I'm clarifying to make sure I understand how this change will impact behavior for users.

My main question here is if a database is using the RFC3339 format for timestamps, will RFC3339Nano be able to properly process the timestamp?

Reply:
Ok, I think I see your concern. To verify if `RFC3339Nano` is able to process data, I changed test data to have mix of `RFC3339Nano` and `RFC3339` [9b50140](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35195/commits/9b50140bef21bbf0315e02fb829f7e51c9b4a269). 

My test still success, and so do other tests.

Here I checked in the debugger, how timestamp is formatted to string, depending which format is used

| Postgres timestamp | t.Format(time.RFC3339Nano) |t.Format(time.RFC3339) |
| ---- | ----- | ----|
| 2022-06-03 21:59:26+00 | 2022-06-03T21:59:26Z | 2022-06-03T21:59:26Z | 
| 2022-06-03 21:59:26.692991+00 | 2022-06-03T21:59:26.692991Z | 2022-06-03T21:59:26Z |
| 2022-06-03 21:59:29.212212+00 | 2022-06-03T21:59:29.212212Z | 2022-06-03T21:59:29Z |
| 2022-06-03 21:59:31+00 | 2022-06-03T21:59:31Z | 2022-06-03T21:59:31Z | 
| 2022-06-03 21:59:31.332121+00 | 2022-06-03T21:59:31.332121Z | 2022-06-03T21:59:31Z |

Reply:
Thanks for clarifying, that's exactly what I was wondering!